modules/mesos/pom.xml

```diff
@@ -76,23 +85,39 @@
                         <phase>generate-sources</phase>
                         <configuration>
                             <target>
-                                <echo message="Update download url in mesos module." />
-                                <echo message="Direct link ${ignite.direct.url}." />
-                                <echo message="Latest link ${ignite.latest.url}." />
+                                <echo message="Update download url in mesos module."/>
+                                <echo message="Latest version ${ignite.version.url}."/>
+                                <echo message="Archive link ${ignite.archive.url}."/>
+
+                                <replaceregexp byline="true" encoding="UTF-8">
+                                    <regexp pattern="(.*IGNITE_LATEST_VERSION_URL = &quot;)(.*)(&quot;.*)"/>
+                                    <substitution expression="\1${ignite.version.url}\3"/>
+                                    <fileset dir="${basedir}/">
+                                        <include name="**/IgniteProvider.java"/>
+                                    </fileset>
+                                </replaceregexp>
+
+                                <replaceregexp byline="true" encoding="UTF-8">
+                                    <regexp pattern="(.*DOWNLOAD_URL_PATTERN = &quot;)(.*)(&quot;.*)"/>
```

Comment:
We can remove this substitution now, I guess. Maybe with ${ignite.archive.url} along. Let's DRY.

Reply:
@alamar Agree. Done.